samples/nrf9160/aws_fota/src/main.c

```diff
@@ -375,16 +378,20 @@ static int client_id_get(char *id_buf, size_t len)
 {
 #if !defined(CONFIG_CLOUD_CLIENT_ID)
 	enum at_cmd_state at_state;
-	char imei_buf[CGSN_RESP_LEN];
+	char imei_buf[IMEI_LEN + 1];
 	int err = at_cmd_write("AT+CGSN", imei_buf, sizeof(imei_buf),
 				&at_state);
 
 	if (err) {
 		printk("Error when trying to do at_cmd_write: %d, at_state: %d",
 			err, at_state);
 	}
-
-	snprintf(id_buf, len, "nrf-%.*s", IMEI_LEN, imei_buf);
+#if !defined(CONFIG_USE_NRF_CLOUD)
```

Comment:
Looks like the error handling is a bit lacking in case `at_cmd_write()` fails, so that `imei_buf` may be used on line 390/392 with random data

Reply:
True, this should probably return.

Reply:
I guess the readout out aws_lib is a little bit more robust and can be reused here: https://github.com/nrfconnect/sdk-nrf/blob/7e940ec1e8a1df2512d87173e7c9b1ee1d5fedbd/subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c#L310-L344 - but I'd say this is out of scope for this PR.

Reply:
Will return in case of error as a simple fix.

Reply:
Thanks, I think that returning `err` is the proper fix here